File: src/client/classes/api/req_res/delete-skill-request.ts
'use strict'

import { GkillAPIRequest } from '../gkill-api-request'

export class DeleteSkillRequest extends GkillAPIRequest {
    name: string
    // path を空にするとスキルを丸ごと消す（画面はこちらだけを使う）
    path: string
    revision: string

    constructor() {
        super()
        this.name = ""
        this.path = ""
        this.revision = ""
    }
}

File: src/server/gkill/api/req_res/delete_skill_request.go
package req_res

type DeleteSkillRequest struct {
	SessionID  string `json:"session_id"`
	LocaleName string `json:"locale_name"`
	Name       string `json:"name"`
	// Path を省くとスキルを丸ごと消す（画面だけが使う）。指定するとそのファイルだけを消す。
	Path string `json:"path"`
	// Revision はファイルを消すときの楽観ロック（省略可）。
	Revision string `json:"revision"`
}
